webapp/channels/src/components/root/root.tsx

```diff
@@ -268,20 +235,14 @@ export default class Root extends React.PureComponent<Props, State> {
                 }
             });
         }
+    };
 
-        if (this.props.location.pathname === '/' && this.props.noAccounts) {
-            this.props.history.push('/signup_user_complete');
-        }
-
+    onConfigLoaded = () => {
         Promise.all([
             this.props.actions.initializeProducts(),
             initializePlugins(),
         ]).then(() => {
-            if (this.mounted) {
-                // supports enzyme tests, set state if and only if
-                // the component is still mounted on screen
-                this.setState({configLoaded: true});
-            }
+            this.setState({shouldMountAppRoutes: true});
```

Comment:
`this.mounted` should still be needed, right? If not `setState` may be called when the object is unmounted, and throw an error.

Reply:
But we are running `initiateMeRequests` right after component did mount. Is the ref mounted still needed then?

Reply:
The problem are the asynchronous tasks. The moment the promise is created, we may be sure the component is mounted. But once we await for the value (or call the then), we don't know how long has passed or what happened in between. One of the things that may happen, is the component getting unmounted.

I guess that since this is the `root` component, we will never unmount it without closing the app, but being purist or if there is any chance the root component gets unmounted, this would still be needed.

Reply:
That is a good to think about. I went in about to find out when do we actually unmount, my first instinct was when the full screen boards is opened. But even that doesnt unmount the `root` component. Then i looked into when this was [added](https://github.com/mattermost/mattermost-webapp/pull/6319/files#diff-504a0deda777621ed3790758a24d4bfecf71fe3974c23472dbd91a7edc25a584R213). It turns out it was added for only the reason to support the enzyme testing.